plugins/provisioners/docker/cap/redhat/docker_install.rb

```diff
@@ -4,10 +4,15 @@ module Cap
       module Redhat
         module DockerInstall
           def self.docker_install(machine)
+            if machine.guest.capability("flavor") == :rhel_8
+              machine.ui.warn("Docker is not supported on RHEL 8 machines.")
```

Comment:
Could probably also provide a link to an article that instructs a user how to get around this. Or could suggest they do something like use the podman provisioner https://github.com/hashicorp/vagrant/issues/11464

Reply:
Borrowing from the get.docker.com script slightly, I think we could say something like:

> RHEL is now only supported by Docker EE. Please install Docker EE or switch to the Podman provisioner to run containers on RHEL.

Does this apply to RHEL < 8 as well?

Reply:
Nice, ya that sounds better!

As far as I can tell, docker isn't supported on RHEL at all. Though the docker docs still have a guide for installing docker ce that works up until (and including) RHEL 7 https://docs.docker.com/install/linux/docker-ce/centos/#install-docker-engine---community.

Maybe the warning should be emitted for all RHEL systems and only hard fail on RHEL 8?

Reply:
I feel like Vagrant should not attempt to install Docker on RHEL:
1. The docker-install script says it's not supported: https://github.com/docker/docker-install/blob/b50fac9805c3d95fb5e8b16d57b19d333e972e20/install.sh#L318
2. [Docker Engine - Community is not supported on Red Hat Enterprise Linux.](https://docs.docker.com/ee/docker-ee/rhel/)

Folks who want to use Docker on RHEL can still install it themselves using a shell provisioner, since this seems to be pretty well documented.